osmosis-types/validation.go

```diff
@@ -34,9 +34,9 @@ func DetermineApplicableTokensInPool(ctx sdk.Context, prKeeper ParticipationRewa
 	pool, _ := ipool.(*participationrewardstypes.OsmosisPoolProtocolData)
 
 	poolDenom := ""
-	for zk, zd := range pool.Zones {
-		if zk == chainID {
-			poolDenom = zd
+	for zk, zd := range pool.Denoms {
```

Comment:
iterating on map called by ```func (k msgServer) Claim()``` 
- this might give rise to non-determinism - can sort the map when its created or before iteration 
- can write a test to verify

Reply:
yep, will make sure it is sorted.